client/apps/edit/components/admin/index.coffee

```diff
@@ -1,29 +1,48 @@
 _ = require 'underscore'
 Backbone = require 'backbone'
+request = require 'superagent'
 async = require 'async'
 featuredListTemplate = -> require('./featured_list.jade') arguments...
 sd = require('sharify').data
 
 module.exports = class EditAdmin extends Backbone.View
 
-  initialize: (options) ->
-    { @article } = options
+  initialize: ({ @article }) ->
     @article.on 'open:tab2', @onOpen
-    @setupAutocomplete()
+    @setupAuthorAutocomplete()
+    @setupFairAutocomplete()
 
-  setupAutocomplete: ->
+  setupAuthorAutocomplete: ->
     Autocomplete = require '../../../../components/autocomplete/index.coffee'
     new Autocomplete
       el: @$('#edit-admin-change-author input')
       url: "#{sd.API_URL}/users?q=%QUERY"
       filter: (res) -> for r in res.results
         { id: r.id, value: r.user.name + ', ' + (r.details?.email or '') }
-      selected: @onAutocompleteSelect
+      selected: @onAuthorSelect
 
-  onAutocompleteSelect: (e, item) =>
+  onAuthorSelect: (e, item) =>
     return unless confirm "Are you sure you want to change the author?"
     @article.trigger('finished').save(author_id: item.id)
 
+  setupFairAutocomplete: ->
+    AutocompleteSelect = require '../../../../components/autocomplete_select/index.coffee'
+    select = AutocompleteSelect @$('#edit-admin-fair .edit-admin-right')[0],
+      url: "#{sd.ARTSY_URL}/api/v1/match/fairs?term=%QUERY"
```

Comment:
Unfortunately no fair resource in v2 at the moment so just directly interacting with v1 here :-/